### PR TITLE
test: Add dynamic allocation port for ganache, fixtures and test dapp

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR addresses an issue encountered when increasing the number of workers to 3 in the CI pipeline. The problem arose because, when increasing the number of jest workers, Detox creates new emulators in the same VM. However, existing fixtures, Ganache, and the test Dapp were not equipped to dynamically manage port allocation, causing test failures due to port conflicts between workers. These failures were particularly noticeable in confirmation tests, as they utilised Ganache, the test Dapp, and different fixtures for each test.

To resolve this issue, this PR introduces a dynamic port allocation mechanism that is deterministic and relies on `process.pid`  instead of `JEST_WORKER_ID` because the worker id is different in the emulator but always `1` outside where servers are spun up. This approach ensures that the number of ports is different between the workers, eliminating port clashes and enabling smooth test execution in multi-worker environments.

~~Included all the confirmation tests in the smoke pattern~~ (removed).

## **Manual testing steps**

No functional change in the code.

## **E2E Pipeline**
The build average time is 17m.

**Before**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/53f3a798-1302-4f91-89f0-c1ca724bee6e
image

**After**
**without confirmations tests**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/6994eecb-bb23-4161-bdac-9a8605334b49
image

**with confirmations tests**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/cd363c8a-8f09-4dd0-b822-132488d68852
image

## **Related issues**

_Fixes https://github.com/MetaMask/MetaMask-planning/issues/1393

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by seaona — DISMISSED** [review]
LGTM! A couple of comments:
- I've created a ticket so we can link this PR to the issue https://github.com/MetaMask/MetaMask-planning/issues/1393
- About Signature tests, I've created a[ separate branch](https://github.com/MetaMask/metamask-mobile/pull/7372) for reorg those tests, in a similar fashion as we have on Extension. This hopefully helps reduce e2e time. I will loop back around that topic

File: e2e/specs/confirmations/send-erc20-with-dapp.spec.js [hunk 1]
```diff
@@ -16,13 +16,10 @@ const HST_CONTRACT = SMART_CONTRACTS.HST;
 const SENT_TOKENS_MESSAGE_TEXT = root.transactions.sent_tokens;
 const WEBVIEW_TEST_DAPP_TRANSFER_TOKENS_BUTTON_ID = 'transferTokens';
 
-describe(Regression('ERC20 tokens'), () => {
+describe(Smoke('ERC20 tokens'), () => {
```
Comment (seaona) [inline e2e/specs/confirmations/send-erc20-with-dapp.spec.js:19]: to discuss with Mobile

File: app/util/test/utils.js [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
 
+export const testConfig = {
+  fixtureServerPort: 12345,
+};
+
 export const isTest =
   process.env.IS_TEST === 'true' &&
   process.env.METAMASK_ENVIRONMENT !== 'production';
+
+export const getFixturesServerPort = () => testConfig.fixtureServerPort;
```
Comment (cryptotavares) [inline app/util/test/utils.js:11]: Why do we need this method? 🤔 
This seems to be static.. Shouldn't we be using this [one](https://github.com/MetaMask/metamask-mobile/pull/7356/files#diff-aec90d924ded2140f8a244ec7298b835b709d093b59de7a2505978c4ab381b97R16-R18)?

File: shim.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
 import { decode, encode } from 'base-64';
+import { isTest, testConfig } from './app/util/test/utils.js';
+import { LaunchArguments } from 'react-native-launch-arguments';
+
+// In a testing environment, assign the fixtureServerPort to use a deterministic port generated by jest workers
+if (isTest) {
+  const raw = LaunchArguments.value();
+  testConfig.fixtureServerPort = raw?.fixtureServerPort
+    ? raw.fixtureServerPort
+    : 12345;
```
Comment (cryptotavares) [inline shim.js:11]: Use exported constant FIXTURE_SERVER_PORT

Comment (vinistevam) [reply shim.js:11]: I can't use the `DEFAULT_FIXTURE_SERVER_PORT` from outside, but I've included one and imported it here.

**Review by seaona — APPROVED** [review]

File: app/util/test/utils.js (hunk 2 above)
Comment (vinistevam) [reply app/util/test/utils.js]: It's been populated on `shim.js`. I renamed this method to avoid confusion with the one outside the `/app`.

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**